packages/react-core/src/components/Select/Select.tsx

```diff
@@ -522,10 +537,12 @@ class Select extends React.Component<SelectProps & InjectedOuiaProps, SelectStat
                   {toggleIcon && <span className={css(styles.selectToggleIcon)}>{toggleIcon}</span>}
                   <span className={css(styles.selectToggleText)}>{placeholderText}</span>
                   {!isCheckboxSelectionBadgeHidden &&
-                    selections &&
-                    (Array.isArray(selections) && selections.length > 0) && (
+                    (customBadgeText !== null ||
```

Comment:
No need to add the `customBadgeText !== null ||` check, you already do that in `generateSelectedBadge`.

Reply:
It's to render the surrounding div and span; otherwise you just get a string back.